### PR TITLE
Connecting Backend to Frontend using AJAX.

### PR DESCRIPTION
**Pull request by yashLadha**
Signed-off-by: Yash Ladha <yashladhapankajladha123@gmail.com>

Fixes #

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- AJAX request to connect backend and frontend services.
-

### CODE REVIEW

**Comment by yashLadha** [comment]
@ParthS007 We can work on this PR. Ping me on gitter for further discussion.

**Comment by yashLadha** [comment]
__WIP__

**Comment by vaibhavsingh97** [comment]
@yashLadha @ParthS007 Can I help?

**Comment by yashLadha** [comment]
With co-authors feature, you can also work. As soon as I get to hostel, I
will add you as a collaborator in my forked copy. Is it ok with you?

**Comment by vaibhavsingh97** [comment]
Sure 👍

**Comment by gabru-md** [comment]
@yashLadha do you guys need any help?
I will be more than happy to be of any assistance.
 :+1:

**Comment by yashLadha** [comment]
@abishekvashok completed, can you remove the label

**Comment by yashLadha** [comment]
@vaibhavsingh97 @gabru-md @ParthS007  Review it.

**Comment by vaibhavsingh97** [comment]
@yashLadha Please check Travsi build Failed

**Comment by yashLadha** [comment]
@vaibhavsingh97 Done

File: frontend/app/templates/index.hbs [hunk 1]
```diff
@@ -1,142 +1,173 @@
...
 <script src="{{rootURL}}js/form.js"></script>
+<script type="text/javascript">
+  $(document).ready(function () {
+    $('#form1').submit(function (event) {
+      event.preventDefault();
+      $.ajaxSetup({
+        headers: {"Access-Control-Allow-Origin": "*"}
+      });
+      $.ajax({
+        url: "http://localhost:5000/api/v1.0/generate_badges",
```
Comment (ankurg22) [inline frontend/app/templates/index.hbs]: Here we would require the link to Heroku app where it is deployed.

File: frontend/app/templates/index.hbs [hunk 2]
```diff
@@ -1,142 +1,173 @@
...
+        data: $(this).serialize(),
+        type: 'POST',
+        success: function (data) {
+          if (data["response"][0]["type"] === "success") {
+            $('#success').css('visibility', 'visible');
+            let link = data["response"][0]["download_link"];
+            link = link.replace("backend/app/", "http://localhost:5000/");
```
Comment (ankurg22) [inline frontend/app/templates/index.hbs]: Same.

File: frontend/app/templates/index.hbs (hunk 1 above)
Comment (yashLadha) [reply frontend/app/templates/index.hbs]: Sure :+1:

File: frontend/app/templates/index.hbs (hunk 2 above)
Comment (yashLadha) [reply frontend/app/templates/index.hbs]: Sure :+1:

**Comment by yashLadha** [comment]
@ankurg22  I will change the url, but the backend again needs to be deployed because there are some changes done in the backend for the connection.

**Comment by abishekvashok** [comment]
@yashLadha pls resolve conflicts. Also you need not worry about backend url as it will auto deployed on heroku I guess.

**Comment by yashLadha** [comment]
Done @abishekvashok

**Comment by abishekvashok** [comment]
Waiting for Travis.

File: frontend/app/templates/index.hbs [hunk 3]
```diff
@@ -1,142 +1,173 @@
...
     </div>
+  </div>
 </div>
 
 <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.0/js/bootstrap.min.js"></script>
 <script src="{{rootURL}}js/jquery.minicolors.min.js"></script>
 <script src="{{rootURL}}js/script.js"></script>
 <script src="{{rootURL}}js/form.js"></script>
+<script type="text/javascript">
+  $(document).ready(function () {
```
Comment (rupav) [inline frontend/app/templates/index.hbs]: Should we open another issue, so that this script can be kept as a separate js file?

Comment (yashLadha) [reply frontend/app/templates/index.hbs]: May be we can reformat the code structure, but for this PR, it is necessary??

**Comment by yashLadha** [comment]
ping @abishekvashok

File: frontend/app/templates/index.hbs (hunk 3 above)
Comment (rupav) [reply frontend/app/templates/index.hbs]: No, this PR don't requires it, LGTM 👍

**Comment by abishekvashok** [comment]
@yashLadha great work but a few concern:
- Nothing happens when we press the generate button.
- All recent upstream changes have been lost. Always when pulling try to keep them. Eg: We dont have check boxes for font. Also another check box is missing. The preview bug was solved but its still present.

Can you resolve these issues?

**Comment by yashLadha** [comment]
As  I mentioned there is a need for updation of backend because some flags need to be added for the functioning, @abishekvashok. and i fetch the upstream.

**Pull request merged**